docs/migrating-v1alpha1-to-v1beta1.md

```diff
@@ -60,11 +59,154 @@ spec:
     value: https://example.com/foo.json
 ```
 
-#### Input and Output PipelineResources
+## PipelineResources and Catalog Tasks
+
+`PipelineResources` are remaining in alpha while the other resource kinds are
+promoted to beta. This is part of a deliberate effort to migrate users away
+from `PipelineResources` onto a broader suite of reusable `Tasks` provided by
+the Tekton catalog.
+
+_More on the reasoning and what's left to do in
+[Why aren't PipelineResources in Beta?](docs/resources.md#why-arent-pipelineresources-in-beta)._
+
+In the near-term we plan to continue to merge fixes to existing `PipelineResource` types, but
+are pushing back against adding significant new features or new `PipelineResource` types, and
+encourage people to use `Tasks` for those instead. At some point the feature may well be deprecated
+and subsequently replaced.
+
+To ease migration away from `PipelineResources`
+[some types have an equivalent `Task` in the Catalog](#pipelineresource-equivalent-catalog-tasks).
+To use these replacement `Tasks` you will need to combine them with your existing `Tasks` via a `Pipeline`.
+
+For example, if you were using this `Task` which was fetching from `git` and building with
+`Kaniko`:
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: build-push-kaniko
+spec:
+  inputs:
+    resources:
+    - name: workspace
+      type: git
+    params:
+    - name: pathToDockerFile
+      description: The path to the dockerfile to build
+      default: /workspace/workspace/Dockerfile
+    - name: pathToContext
+      description: The build context used by Kaniko
+      default: /workspace/workspace
+  outputs:
+    resources:
+    - name: builtImage
+      type: image
+  steps:
+  - name: build-and-push
+    image: gcr.io/kaniko-project/executor:v0.17.1
+    env:
+    - name: "DOCKER_CONFIG"
+      value: "/tekton/home/.docker/"
+    args:
+    - --dockerfile=$(inputs.params.pathToDockerFile)
+    - --destination=$(outputs.resources.builtImage.url)
+    - --context=$(inputs.params.pathToContext)
+    - --oci-layout-path=$(inputs.resources.builtImage.path)
+    securityContext:
+      runAsUser: 0
+```
+
+_Note that using `PipelineResources` with this `Task` has an immediate downside in that it is
+coupled to `git`; one can't use the same `Task` with files that were obtained some other way._
```

Comment:
I'm not sure that this is completely - we do support optional pipeline resources - so a Task that supports an optional pipeline resource could also accept a parameter that points to the location of the data if the resource is not provided. We probably would need to improve of this to make it nice and easy to use.

Reply:
Right, but this would be band-aid.

Reply:
hm interesting point - I hadn't thought about optional PipelineResources! I think having the git coupling in the Task is still not the greatest, but like you said it's not actually _required_ anymore - I'm going to just remove this sentence for now!